lib/hal/impression/rpihal.ex

```diff
@@ -136,20 +165,22 @@ defmodule Inky.Impression.RpiHAL do
     Logger.info(msg)
   end
 
-  defp log(_state, msg) when is_binary(msg) do
+  defp log(state, msg) when is_binary(msg) do
     IO.puts(msg)
     Logger.info(msg)
     state
   end
 
   defp do_update(state, display, border, buffer) do
-    IO.inspect(buffer, label: "buffer (rpihal.ex:138)")
     Logger.info("border: #{inspect(border)}")
-    border = :red
+    border = :black
+    d_pd = display.packed_dimensions
 
     state
     |> log("setting resolution")
     |> set_resolution(display.packed_resolution)
+    |> log("setting dimensions")
+    |> set_dimensions(d_pd.width, d_pd.height)
```

Comment:
Here I've added `set_dimensions` after `set_resolution`. The result is the same with either one of the two set or both being active.